### ISSUE TITLE
Ultra-slow on raspberry pi

### ISSUE THREAD

**Issue by IngwiePhoenix** [Trixarian/sakis3g-source#5]
Hey there!

I am currently going by the instructions given in the readme. But whenever I launch Sakis - it takes forever to at least display one thing.

Looking into the file, gives away that you are putting a gzipped version of usbmodeswitch into the file - and what else? Is there a way to use an existing set of binaries, like usbmodesiwtch? Because I want to use sakis3g in a script, and I dont want to have this script run forever x)

Any way of optimizing the speed would be great. BTW, I am using latest Raspbian.

**Comment by Trixarian** [comment]
Yes and no. The default compile creates a script with the usb_modeswitch binary and database embedded into it, but a little known feature of the compile script is that you can also use it to generate a script with ONLY the usb_modeswitch binary in it or just the script without anything embedded into it. It's as simple as using `./compile embedded` or `./compile stripped` to get each of the other options respectively. The stripped version is the one you're looking for.

But there's a problem. Even if you install usb_modeswitch separately, it won't work with sakis3g in it's current form. You're going to have to overwrite the usb_modeswitch database installed by your distro with the files found in sakis3g-source/dependencies/usb-modeswitch-data before sakis3g will be able to use it to switch your device. Once that's done, it should work without too much effort - especially if the usb_modeswitch binary is optimized for hard float.

Mind you - it will still take between 30 second to 90 second to fully connect you. On an old P3 it takes me 90 seconds to connect using sakis3g, while on my newer P4 it only takes about 30 seconds with configuration from the menu. It really depends on the system.

**Comment by IngwiePhoenix** [comment]
Hey.

Thanks for the reply, I will try this!

The good part is: usb_modesswitch has my usb already in the known list - it switches it properly already. (i get wwan0 by just plugging the thing in, /var/log/messages reports a proper usb switching). Therefore, do I still need to modify?

I am not concernted about connecting, but rather about launching the script itself. For a moment, I had thought the script simply didnt want to work on my Pi, but then suddenly the UI came up….oh. x)

I will try the stripped version later this night! ^^

By the way, do you know how to change the current active gateway? Like, when I connect using sakis3g, and have ppp0 being the currently prefered device, how would i temporarily switch back to wlan0? Currently, I disconnect ppp0 and re-plug in my usb wifi dongle, then it magically would work over wifi again.
Any good tipp for that one?
Am 18.03.2014 um 21:42 schrieb Brenton Scott notifications@github.com:

**Comment by Trixarian** [comment]
A few ways comes to mind, but none of them are as simple and straight forward as just swapping devices :unamused:

It's good that usb_modeswitch is switching your device automatically - you can just use one of the other options in the script and connect without switching the device first.

**Comment by Trixarian** [comment]
Looks like I was wrong. From my tests, I've seen that on systems that has usb_modeswitch installed, sakis3g will prefer it over it's internal one much like it will prefer wvdial over ppp if found. It will also be able to switch the devices using the distro's usb_modeswitch-data without any problems.
